models/classes/Platform/Repository/Lti1p3RegistrationRepository.php

```diff
@@ -136,21 +136,28 @@ private function createRegistrationByProvider(LtiProvider $ltiProvider): ?Regist
             ->getKeyChains());
 
         $platformKeyChain = current($this->getPlatformKeyChainRepository()
-            ->findAll(new KeyChainQuery())
+            ->findAll(new KeyChainQuery($ltiProvider->getId()))
             ->getKeyChains());
 
-        if ($toolKeyChain === false || $platformKeyChain === false) {
+        if ($platformKeyChain === false) {
             return null;
         }
 
+        $translatedToolKeyChain = null;
+        if ($toolKeyChain !== false && empty($ltiProvider->getToolJwksUrl())) {
+            $translatedToolKeyChain = $this->translateKeyChain($toolKeyChain);
+        }
+
         return new Registration(
             $ltiProvider->getId(),
             $ltiProvider->getToolClientId(),
             $this->getDefaultPlatform(),
             $this->getTool($ltiProvider),
             $ltiProvider->getToolDeploymentIds(),
             $this->translateKeyChain($platformKeyChain),
-            $this->translateKeyChain($toolKeyChain)
+            $translatedToolKeyChain,
+            $this->getOption(self::OPTION_ROOT_URL) . 'taoLti/Security/jwks',
```

Comment:
WDYT about moving `taoLti/Security/jwks` to the constants?

Reply:
good suggestion